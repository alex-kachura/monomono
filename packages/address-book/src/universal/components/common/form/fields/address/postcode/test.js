import React from 'react';
import { render } from 'react-testing-library';
import { renderProviders } from '@oneaccount/react-foundations/lib/test-utils';
import { DefaultThemeProvider } from '@beans/theme';
import Postcode from './';
import { Formik } from '@oneaccount/react-foundations';

const postcodeProps = {
  id: 'postcode',
  name: 'postcode',
  label: 'Postcode',
  placeholder: 'e.g. EC1R 2NT',
  required: true,
  handleFindAddress: jest.fn(),
};

function renderPostcode({ value = '', error = null, fetchFn }) {
  return render(
    renderProviders({
      appConfig: {
        getLocalePhrase: (key) => key,
        fetch: fetchFn,
      },
      children: (
        <DefaultThemeProvider>
          <Formik initialValues={{ postcode: value }} initialErrors={{ postcode: error }}>
            <Postcode {...postcodeProps} />
          </Formik>
        </DefaultThemeProvider>
      ),
    }),
  );
}

describe('[Component: Postcode]', () => {
  describe('[State: Default]', () => {
    it('should render correctly', () => {
      const component = renderPostcode({ value: '' });

      expect(component.asFragment()).toMatchSnapshot();
    });
  });

  describe('[State: Initial Value]', () => {
    it('should render correctly', () => {
      const component = renderPostcode({
        value: 'EC1R 2NT',
      });

      expect(component.asFragment()).toMatchSnapshot();
    });
  });

  describe('[State: Error]', () => {
    it('should render correctly', () => {
      const component = renderPostcode({
        value: 'EC1R 2NT',
        error: 'Error on field',
      });

      expect(component.asFragment()).toMatchSnapshot();
    });
  });
});
