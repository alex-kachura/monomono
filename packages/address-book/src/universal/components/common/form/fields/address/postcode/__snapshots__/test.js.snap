// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: Postcode] [State: Default] should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-htpNat fhaZgK styled__Container-sc-1bf1p22-0 hTDNaU"
  >
    <label
      class="styled__StyledLabel-sc-28yvb4-0 bPkari"
      for="postcode"
    >
      Postcode 
    </label>
    <div
      class="base-components__BaseElement-sc-150tvch-0 beans-form-group__children styled__FormGroupChildren-wt4ift-2 bNvdOT"
    >
      <input
        aria-disabled="false"
        class="styled__Container-sc-1bf1p22-0 eZwXEO"
        id="postcode"
        name="postcode"
        placeholder="e.g. EC1R 2NT"
        type="text"
        value=""
      />
      <button
        aria-disabled="false"
        class="styled__BaseButton-sc-3mnirm-0 find-address styled__PrimaryButton-sc-3mnirm-2 giYBMC"
        role="button"
        type="button"
      >
        <span
          class="base-components__BaseElement-sc-150tvch-0 styled__InnerContainer-sc-3mnirm-1 fUwmia"
        >
          <span
            class="base-components__BaseElement-sc-150tvch-0 styled__BaseSpan-sc-3mnirm-4 beans-button__text styled__TextSpan-sc-3mnirm-5 eqKhmg"
          >
            address.find-address
          </span>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`[Component: Postcode] [State: Error] should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-htpNat fhaZgK styled__Container-sc-1bf1p22-0 hTDNaU"
  >
    <label
      class="styled__StyledLabel-sc-28yvb4-0 BlFly"
      for="postcode"
    >
      Postcode 
    </label>
    <div
      class="base-components__BaseElement-sc-150tvch-0 beans-form-group__children styled__FormGroupChildren-wt4ift-2 bNvdOT"
    >
      <input
        aria-describedby="postcode-error"
        aria-disabled="false"
        class="styled__Container-sc-1bf1p22-0 hIxSdw"
        id="postcode"
        name="postcode"
        placeholder="e.g. EC1R 2NT"
        type="text"
        value="EC1R 2NT"
      />
      <button
        aria-disabled="false"
        class="styled__BaseButton-sc-3mnirm-0 find-address styled__PrimaryButton-sc-3mnirm-2 eQEoWE"
        role="button"
        type="button"
      >
        <span
          class="base-components__BaseElement-sc-150tvch-0 styled__InnerContainer-sc-3mnirm-1 fUwmia"
        >
          <span
            class="base-components__BaseElement-sc-150tvch-0 styled__BaseSpan-sc-3mnirm-4 beans-button__text styled__TextSpan-sc-3mnirm-5 eqKhmg"
          >
            address.find-address
          </span>
        </span>
      </button>
    </div>
    <p
      class="base-components__BaseElement-sc-150tvch-0 beans-form-group__message beans-form-group__error-message styled__Message-wt4ift-3 jcJuKh typography___styledElement-sc-186l0ce-0 dmJyuB"
      id="postcode-error"
    >
      Error on field
    </p>
  </div>
</DocumentFragment>
`;

exports[`[Component: Postcode] [State: Initial Value] should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-htpNat fhaZgK styled__Container-sc-1bf1p22-0 hTDNaU"
  >
    <label
      class="styled__StyledLabel-sc-28yvb4-0 bPkari"
      for="postcode"
    >
      Postcode 
    </label>
    <div
      class="base-components__BaseElement-sc-150tvch-0 beans-form-group__children styled__FormGroupChildren-wt4ift-2 bNvdOT"
    >
      <input
        aria-disabled="false"
        class="styled__Container-sc-1bf1p22-0 eZwXEO"
        id="postcode"
        name="postcode"
        placeholder="e.g. EC1R 2NT"
        type="text"
        value="EC1R 2NT"
      />
      <button
        aria-disabled="false"
        class="styled__BaseButton-sc-3mnirm-0 find-address styled__PrimaryButton-sc-3mnirm-2 giYBMC"
        role="button"
        type="button"
      >
        <span
          class="base-components__BaseElement-sc-150tvch-0 styled__InnerContainer-sc-3mnirm-1 fUwmia"
        >
          <span
            class="base-components__BaseElement-sc-150tvch-0 styled__BaseSpan-sc-3mnirm-4 beans-button__text styled__TextSpan-sc-3mnirm-5 eqKhmg"
          >
            address.find-address
          </span>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
